actions/userinfo.js

```diff
@@ -1,17 +1,218 @@
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer ' + localStorage.getItem('id_token')
+      }
+    }).then(resp => {
+      return {
+        email: resp.data.email,
+        user_id: resp.data.user_id,
+        user_metadata: resp.data.user_metadata
+      };
+    });
+  };
+}
+
+export function resetPassword(email) {
+  return dispatch => {
+    const payload = {
+      client_id: 'OIzT7gmLYcxbLZWGDz7LAsX6i2iCP2tc',
+      email: email,
+      connection: 'Username-Password-Authentication'
+    };
+    return Axios.post(
+      'https://gww.auth0.com/dbconnections/change_password',
+      payload,
+      {
+        headers: { 'content-type': 'application/json' }
+      }
+    ).then(resp => {
+      swal(resp.data, null, 'success');
+      return dispatch(resetPasswordEmailSent());
+    });
+  };
+}
+function resetPasswordEmailSent() {
+  return {
+    type: 'RESET_PASSWORD_EMAIL_SENT'
+  };
+}
+export function logout() {
+  localStorage.clear();
+  browserHistory.replace('/dashboard');
   return {
-    type: USER_ID,
-    id,
+    type: LOGOUT_SUCCESS
+  };
+}
+export function isAuthenticated() {
+  let expiresAt = JSON.parse(localStorage.getItem('expires_at'));
+  return new Date().getTime() < expiresAt;
+}
+function getAccessToken() {
+  const accessToken = localStorage.getItem('access_token');
+  if (!accessToken) {
+    return null;
+  }
+  return accessToken;
+}
+function getUserMetadata(userId) {
+  return Axios.get(`https://gww.auth0.com/api/v2/users/${userId}`, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: 'Bearer ' + localStorage.getItem('id_token')
+    }
+  }).then(resp => {
+    return resp.data.user_metadata;
+  });
+}
+export function getUserInfo() {
+  let accessToken = getAccessToken();
+  return (dispatch, getState) => {
+    return new Promise((res, rej) => {
+      if (!accessToken) {
+        return local.post('/createUser').then(resp => {
+          return dispatch(loginSuccess(resp.data));
+        });
+      }
+      lock.getUserInfo(accessToken, (error, profile) => {
+        if (error) {
+          return rej(error);
+        }
+        return res(profile);
+      });
+    })
+      .then(profile => {
+        return getUserMetadata(profile.sub).then(meta => {
+          if (!meta) {
+            return updateUser({}, profile.sub);
+          }
+          return {
+            user_metadata: meta,
+            email: profile.email,
+            user_id: profile.sub
+          };
+        });
+      })
+      .then(profile => {
+        return local
+          .post('/createUser', { auth0_id: profile.user_id })
+          .then(res => {
+            return dispatch(
+              loginSuccess({
+                ...profile,
+                ...res.data
+              })
+            );
+          });
+      })
+      .catch(error => {
+        return dispatch(loginError(error));
+      });
   };
```

Comment:
I would bring these Axios calls to the update the user, reset the password all into the auth module that was previously created.

Leave this file as a redux action creator and dispatcher, move the heavy lifting work of these auth0 api calls into the auth module.

Reply:
check